### PR TITLE
core(emulation): use Moto G4 for mobile emulation

### PR DESCRIPTION
**Pull request by connorjclark**
part of #9876

### CODE REVIEW

File: lighthouse-core/lib/emulation.js [hunk 1]
```diff
@@ -8,19 +8,18 @@
...
-  height: 660,
+  screenWidth: 360,
+  screenHeight: 640,
+  width: 360,
+  height: 640,
   positionX: 0,
   positionY: 0,
   scale: 1,
-  deviceScaleFactor: 2.625,
+  deviceScaleFactor: 2.625, // ??
```
Comment (connorjclark) [inline lighthouse-core/lib/emulation.js]: what the heck is this @paulirish

File: lighthouse-core/lib/emulation.js [hunk 2]
```diff
@@ -8,19 +8,18 @@
...
  * @type {LH.Crdp.Emulation.SetDeviceMetricsOverrideRequest}
  */
-const NEXUS5X_EMULATION_METRICS = {
+const MOTOG4_EMULATION_METRICS = {
   mobile: true,
-  screenWidth: 412,
-  screenHeight: 660,
-  width: 412,
-  height: 660,
+  screenWidth: 360,
```
Comment (connorjclark) [inline lighthouse-core/lib/emulation.js]: Need to look at an actual device and see what a browser reports this as.

File: lighthouse-core/lib/emulation.js [hunk 3]
```diff
@@ -161,6 +160,6 @@ module.exports = {
   enableCPUThrottling,
   disableCPUThrottling,
   goOffline,
-  MOBILE_USERAGENT: NEXUS5X_USERAGENT,
+  MOBILE_USERAGENT: MOTOG4_USERAGENT,
```
Comment (connorjclark) [inline lighthouse-core/lib/emulation.js]: nice thinking here on the export

File: lighthouse-core/test/report/html/renderer/report-renderer-test.js [hunk 4]
```diff
@@ -197,10 +197,10 @@ describe('ReportRenderer', () => {
       assert.ok(descriptions.length >= 3);
 
       const descriptionsTxt = descriptions.map(el => el.textContent).join('\n');
-      assert.ok(/Nexus/.test(descriptionsTxt), 'should have added device emulation');
-      assert.ok(/RTT/.test(descriptionsTxt), 'should have added network');
-      assert.ok(/\dx/.test(descriptionsTxt), 'should have added CPU');
-      assert.ok(descriptionsTxt.includes(sampleResults.userAgent), 'user agent populated');
+      expect(descriptionsTxt).toContain('Moto G4');
```
Comment (connorjclark) [inline lighthouse-core/test/report/html/renderer/report-renderer-test.js]: asserts are bad.

File: lighthouse-core/lib/emulation.js (hunk 1 above)
Comment (patrickhulce) [reply lighthouse-core/lib/emulation.js]: that should be the device DPI

Comment (patrickhulce) [reply lighthouse-core/lib/emulation.js]: looks like it's 3 for moto g4 

https://github.com/WPO-Foundation/webpagetest/blob/901d2906711c32577f9c8e19603cf491096fcbf8/www/settings/mobile_devices.ini#L9

File: lighthouse-core/lib/emulation.js (hunk 2 above)
Comment (patrickhulce) [reply lighthouse-core/lib/emulation.js]: 360 looks right https://github.com/WPO-Foundation/webpagetest/blob/901d2906711c32577f9c8e19603cf491096fcbf8/www/settings/mobile_devices.ini#L7

could use a WPT run if you need to to verify

File: lighthouse-core/lib/emulation.js [hunk 5]
```diff
@@ -39,7 +38,7 @@ const DESKTOP_EMULATION_METRICS = {
 };
 
 // eslint-disable-next-line max-len
-const NEXUS5X_USERAGENT = 'Mozilla/5.0 (Linux; Android 6.0.1; Nexus 5 Build/MRA58N) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/80.0.3963.0 Mobile Safari/537.36 Chrome-Lighthouse';
+const MOTOG4_USERAGENT = 'Mozilla/5.0 (Linux; Android 6.0.1; Moto G (4)) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/80.0.3963.0 Mobile Safari/537.36 Chrome-Lighthouse';
```
Comment (patrickhulce) [inline lighthouse-core/lib/emulation.js]: fwiw WPT adds `Build/MPJ24.139-64`

Comment (connorjclark) [reply lighthouse-core/lib/emulation.js]: hmm... I copied this from WPT

Comment (patrickhulce) [reply lighthouse-core/lib/emulation.js]: like from a real phone WPT run? maybe the emulated UA is a little off 😆

File: lighthouse-core/lib/emulation.js (hunk 1 above)
Comment (patrickhulce) [reply lighthouse-core/lib/emulation.js]: math checks out too `640 * 3 = 1920`, `360 * 3 = 1080`

File: lighthouse-core/lib/emulation.js (hunk 5 above)
Comment (connorjclark) [reply lighthouse-core/lib/emulation.js]: https://www.webpagetest.org/result/200115_VW_c15479a68affb254d3322ce865b0adf7/

click to the LH report and scroll to the bottom. I grabbed the host UA.

Comment (patrickhulce) [reply lighthouse-core/lib/emulation.js]: yep I guess the WPT emulated UA is a bit off then!

Comment (connorjclark) [reply lighthouse-core/lib/emulation.js]: I assumed this is not emulating anything.

image


How are you getting the UA w/ `Build/MPJ24.139-64`

Comment (patrickhulce) [reply lighthouse-core/lib/emulation.js]: You are correct, the emulated one mistakenly contains `Build/MPJ24.139-64`

https://github.com/WPO-Foundation/webpagetest/blob/901d2906711c32577f9c8e19603cf491096fcbf8/www/settings/mobile_devices.ini#L11

File: lighthouse-cli/test/smokehouse/test-definitions/byte-efficiency/expectations.js [hunk 6]
```diff
@@ -169,12 +169,12 @@ const expectations = [
           },
         },
         'uses-responsive-images': {
-          displayValue: 'Potential savings of 69\xa0KB',
+          displayValue: 'Potential savings of 53\xa0KB',
```
Comment (connorjclark) [inline lighthouse-cli/test/smokehouse/test-definitions/byte-efficiency/expectations.js]: these values are calculated  based on DPR. They are proportional to `DPR^2`.

`3^2 / 2.625^2 = ~1.3`, so I divided each expectation by 1.3

**Pull request merged**